mapper/orm-coordination-outbox-polling/src/main/java/org/hibernate/search/mapper/orm/coordination/outboxpolling/event/impl/OutboxPollingOutboxEventAdditionalJaxbMappingProducer.java

```diff
@@ -157,50 +112,60 @@ public Collection<MappingDocument> produceMappings(ConfigurationPropertySource p
+		}
+		else {
+			int resolvedUuidType = UuidDataTypeUtils.uuidType(
+					uuidType.orElse(
+							HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_OUTBOX_EVENT_UUID_TYPE ),
+					dialect,
+					ENTITY_MAPPING_OUTBOXEVENT_UUID_TYPE.resolveOrRaw( propertySource )
+			);
+			String resolvedUuidStrategy = uuidStrategy.orElse(
+					HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_OUTBOX_EVENT_UUID_GEN_STRATEGY )
+					.strategy();
+			mappings = createMappings(
+					schema.orElse( "" ),
+					catalog.orElse( "" ),
+					table.orElse(
+							HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_OUTBOX_EVENT_TABLE ),
+					resolvedUuidType, resolvedUuidStrategy,
+					isDiscriminatorMultiTenancyEnabled( buildingContext )
+			);
 		}
 
-		String resolvedUuidType = UuidDataTypeUtils.uuidType(
-				uuidType.orElse(
-						HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_OUTBOX_EVENT_UUID_TYPE ),
-				dialect );
-
-		@SuppressWarnings("deprecation")
-		String entityDefinition = mapping.orElseGet( () -> String.format(
-				Locale.ROOT,
-				ENTITY_DEFINITION_TEMPLATE,
-				schema.orElse( "" ),
-				catalog.orElse( "" ),
-				table.orElse(
-						HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_OUTBOX_EVENT_TABLE ),
-				uuidStrategy.orElse(
-						HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_OUTBOX_EVENT_UUID_GEN_STRATEGY )
-						.strategy(),
-				resolvedUuidType,
-				PayloadMappingUtils.payload(
-						payloadType.orElse(
-								HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_OUTBOX_EVENT_PAYLOAD_TYPE ),
-						false
-				)
-		) );
-
-		log.outboxEventGeneratedEntityMapping( entityDefinition );
-		Origin origin = new Origin( SourceType.OTHER, HIBERNATE_SEARCH );
-
-		ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream( entityDefinition.getBytes() );
-		BufferedInputStream bufferedInputStream = new BufferedInputStream( byteArrayInputStream );
-		Binding<?> binding = mappingBinder.bind( bufferedInputStream, origin );
-
-		JaxbHbmHibernateMapping root = (JaxbHbmHibernateMapping) binding.getRoot();
-
-		MappingDocument mappingDocument = new MappingDocument( HIBERNATE_SEARCH, root, origin, buildingContext );
-		return Collections.singletonList( mappingDocument );
+		log.outboxEventGeneratedEntityMapping( mappings );
+
+		return Map.of( OutboxEvent.class, mappings );
+	}
+
+	private static JaxbEntityMappings createMappings(String schema, String catalog,
+			String table, int resolvedUuidType, String resolvedUuidStrategy,
+			boolean tenantIdRequired) {
+		AdditionalMappingBuilder builder = new AdditionalMappingBuilder(
+				OutboxEvent.class, ENTITY_NAME )
+				.id( resolvedUuidType, resolvedUuidStrategy )
+				.index( "entityIdHash" )
+				.index( "status" )
+				.index( "processAfter" )
```

Comment:
Just to be sure: you did check that the resulting schema on the database side remains the same, right? I wonder about these indexes in particular, they might have a different name as they are defined slightly differently.

Reply:
Yes, when I was looking at the column types I've also looked at the queries generated for creating indexes -- and those looked the same (same index names)